detect_git_test.go

```diff
@@ -0,0 +1,58 @@
+package getter
+
+import (
+	"testing"
+)
+
+func TestGitDetector(t *testing.T) {
+	cases := []struct {
+		Input  string
+		Output string
+	}{
```

Comment:
Some systems like AWS code commit use non git user names: `git clone ssh://Your-SSH-Key-ID@git-codecommit.us-east-2.amazonaws.com/v1/repos/MyDemoRepo my-demo-repo`, not sure if its worth handling that in SCP style

Reply:
Yeah, for now you have just force it with `git::ssh://`, the autodetection won't work for custom usernames. We can relax this later but I wanted to limit blast radius to start, and there is a decent workaround as noted 😄